File: Cores/OSOL.Extremum.Core.DotNet/Random/Distributions/ContinuousUniform.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace OSOL.Extremum.Core.DotNet.Random.Distributions
{
    public interface ContinuousUniform
    {
        double getContinuousUniform(double min, double max);   
    }

    public static class ContinuousUniformFunctions
    {
        public static Dictionary<string, double> Diameter(this ContinuousUniform GoRN, Dictionary<string, Tuple<double, double>> area) =>
            area.ToDictionary(kvp => kvp.Key,
                kvp => GoRN.getContinuousUniform(kvp.Value.Item1, kvp.Value.Item2));
    }
}

File: Cores/JVM/src/main/scala/OSOL/Extremum/Cores/JVM/Random/Distributions/ContinuousUniform.scala
package OSOL.Extremum.Cores.JVM.Random.Distributions

trait ContinuousUniform {

  def getContinuousUniform(min: java.lang.Double, max: java.lang.Double): java.lang.Double

  final def getContinuousUniform(area: Map[String, (java.lang.Double, java.lang.Double)]): Map[String, java.lang.Double] =
    area.map { case (key, (min, max)) => (key, getContinuousUniform(min, max)) }

  def getContinuousUniformScala(min: Double, max: Double): Double =
    getContinuousUniform(min, max).doubleValue()

  final def getContinuousUniformScala(area: Map[String, (Double, Double)]): Map[String, Double] =
    area.map { case (key, (min, max)) => (key, getContinuousUniformScala(min, max)) }


}